packages/ejector/crates/utils/src/eject.rs

```diff
@@ -0,0 +1,100 @@
+use std::time::Duration;
+
+use alloy::{
+    primitives::{Address, U256},
+    providers::{Provider, ProviderBuilder},
+    transports::http::reqwest::Url,
+};
+use tracing::{info, warn};
+
+use crate::receipt::poll_receipt_until;
+
+use super::lookahead::Lookahead;
+
+pub async fn eject_operator(
+    l1_http_url: Url,
+    signer: alloy::signers::local::PrivateKeySigner,
+    whitelist_addr: Address,
+    lookahead: Lookahead,
+    min_operators: u64,
+) -> eyre::Result<()> {
+    let l1 = ProviderBuilder::new().wallet(signer.clone()).connect_http(l1_http_url.clone());
+
+    let preconf_whitelist = bindings::IPreconfWhitelist::new(whitelist_addr, l1.clone());
+
+    let active_operators = active_operator_count(&preconf_whitelist).await?;
+
+    if min_operators > 0 && active_operators <= min_operators {
+        warn!(
+            "Not ejecting operator: operator_count {}, min_operators {}",
+            active_operators, min_operators
+        );
+
+        return Ok(());
+    }
+
+    let operator = match lookahead {
+        Lookahead::Current => preconf_whitelist.getOperatorForCurrentEpoch().call().await?,
+        Lookahead::Next => preconf_whitelist.getOperatorForNextEpoch().call().await?,
+    };
+
+    info!("Ejecting operator: {operator:#x} for lookahead: {lookahead:?}");
+
+    if operator.is_zero() {
+        warn!("{lookahead:?} operator is zero; skipping eject.");
+        return Ok(());
+    }
+
+    let pending = preconf_whitelist.removeOperator(operator, true).send().await?;
```

Comment:
do we need to add a retry strategy here? like exponential backoff or whatever we do for other stuff in taiko-client @cyberhorsey

Reply:
eh probably but its good enoug hfor a first draft i think and can be added to in further PRs